dockstore-webservice/src/main/java/io/dockstore/webservice/languages/NextflowHandler.java

```diff
@@ -102,12 +103,28 @@ public Version parseWorkflowContent(String filepath, String content, Set<SourceF
             } else {
                 createValidationForMissingMainScript(version, filepath, mainScriptPath);
             }
+            updateDescriptorTypeVersion(sourceFiles, version, configuration, potentialScript);
         } catch (NextflowUtilities.NextflowParsingException e) {
             createValidationForGeneralFailure(version, filepath);
         }
         return version;
     }
 
+    /**
+     *  Sets source files' type version, and updates the workflow version descriptor type version
+     *  based on the source files's descriptor type version.
+     * @param sourceFiles
+     * @param version
+     * @param configuration
+     * @param mainScript
+     */
+    private void updateDescriptorTypeVersion(final Set<SourceFile> sourceFiles, final Version version,
+            final Configuration configuration, final Optional<SourceFile> mainScript) {
+        final String dslVersion = this.calculateDslVersion(configuration, mainScript).orElse(null);
+        sourceFiles.forEach(sourceFile -> sourceFile.setTypeVersion(dslVersion));
```

Comment:
Does this set the version for all source files, including non-descriptor files like test parameter files? I think we only want to set it for descriptor files

Reply:
Ah, good catch!

Reply:
(I vaguely recall someone from nf-core mentioning that they were working on it)